prometheus/histogram.go

```diff
@@ -1835,3 +1835,129 @@ func (n *nativeExemplars) addExemplar(e *dto.Exemplar) {
 		n.exemplars = append(n.exemplars[:nIdx], append([]*dto.Exemplar{e}, append(n.exemplars[nIdx:rIdx], n.exemplars[rIdx+1:]...)...)...)
 	}
 }
+
+type constNativeHistogram struct {
+	desc                            *Desc
+	count                           uint64
+	sum                             float64
+	labelPairs                      []*dto.LabelPair
+	nativeHistogramSchema           int32
+	nativeHistogramZeroThreshold    float64
+	nativeHistogramMaxZeroThreshold float64
+	nativeHistogramMaxBuckets       uint32
+	nativeHistogramMinResetDuration time.Duration
+	timeStamp                       time.Time
+	nativeExemplars                 []*dto.Exemplar
+
+	positiveBuckets map[int]int64
+	negativeBuckets map[int]int64
+	zeroBucket      uint64
+}
+
+func NewconstNativeHistogram(desc *Desc, count uint64, sum float64, postiveBuckets, negativeBuckets map[int]int64, zeroBucket uint64,
+	labelPairs []*dto.LabelPair, nativeHistogramSchema int32, nativeHistogramZeroThreshold float64,
+	nativeHistogramMaxZeroThreshold float64, nativeHistogramMaxBuckets uint32,
+	nativeHistogramMinResetDuration time.Duration,
+	timeStamp time.Time,
+	nativeExemplars []*dto.Exemplar,
+) constNativeHistogram {
```

Comment:
```suggestion
) (constNativeHistogram, error) {
```

I believe we need to do some validation before returning the object. The first thing I can remember is validating labels, but maybe there are more stuff we need to validate?